### PR TITLE
Stats calculator unit tests framework + example for OutputNode

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorAssertion.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        for (int sourceId = 0; sourceId < planNode.getSources().size(); ++sourceId) {
+            return withSourceStats(sourceId, sourceStatsBuilderConsumer);
+        }
+        return this;
+    }
+
+    public StatsCalculatorAssertion withSourceStats(PlanNodeStatsEstimate sourceStats)
+    {
+        checkPlanNodeSet();
+        for (int sourceId = 0; sourceId < planNode.getSources().size(); ++sourceId) {
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorAssertion.java]: shouldn't there be just one source in this case?

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        this.session = queryRunner.getDefaultSession();
+        this.metadata = queryRunner.getMetadata();
+    }
+
+    private static LocalQueryRunner createQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java]: fix comment `parallel`

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        this.statsCalculator = queryRunner.getStatsCalculator();
+        this.session = queryRunner.getDefaultSession();
+        this.metadata = queryRunner.getMetadata();
+    }
+
+    private static LocalQueryRunner createQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("local")
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java]: don't add tpch connector if not needed

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java [hunk 4]
```diff
@@ -405,6 +406,54 @@ public WindowNode window(WindowNode.Specification specification, Map<Symbol, Win
...
+    {
+        OutputBuilder outputBuilder = new OutputBuilder();
+        outputBuilderConsumer.accept(outputBuilder);
+        return outputBuilder.build();
+    }
+
+    public class OutputBuilder
+    {
+        private PlanNode source;
+        private List<String> columnNames = new ArrayList<>();
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java:430]: use ImmutableList.Builder

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java [hunk 5]
```diff
@@ -405,6 +406,54 @@ public WindowNode window(WindowNode.Specification specification, Map<Symbol, Win
...
+            return this;
+        }
+
+        protected OutputNode build()
+        {
+            return new OutputNode(idAllocator.getNextId(), source, columnNames, outputs);
+        }
+    }
+
+    public OutputNode output(PlanNode source, List<String> columnNames, List<Symbol> symbols)
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: move below other `output` function?

**Review by sopel39 — APPROVED** [review]

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorAssertion.java (hunk 1 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorAssertion.java]: Done 3fbd1a0

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java (hunk 3 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java]: Done b7dd6e9

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java (hunk 2 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java]: DOne b7dd6e9

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java (hunk 4 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java:430]: Let's leave it. It is consistent with other builders and is converted to immutable collection anyway as soon as plan node is built

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java (hunk 5 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: Done 2eb858b

**Comment by losipiuk** [comment]
Merged. Thanks!

**Pull request closed without merging**